### PR TITLE
support setting annotations on datadog-operator deployment

### PR DESCRIPTION
**Pull request by 0robustus1**
#### What this PR does / why we need it:

In our deployment of datadog-operator we want to utilize https://github.com/stakater/Reloader to automatically restart (a.k.a. perform a fresh rollout) the deployment of the operator when secrets (e.g. api-key, app-key) change.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [x] `CHANGELOG.md` has been updated
- [x] Variables are documented in the `README.md`
- [x] For Datadog Operator chart or value changes update the test baselines (run: `make update-test-baselines`)

### CODE REVIEW

**Comment by 0robustus1** [comment]
fyi i was not able to run `make update-test-baselines` successfully. It also fails on the current `main` branch for me.

**Review by clamoriniere — APPROVED** [review]

**Comment by clamoriniere** [comment]
> fyi i was not able to run `make update-test-baselines` successfully. It also fails on the current `main` branch for me.

thanks for the information. I have the same issue, we will fix it in on main soon.

File: charts/datadog-operator/templates/deployment.yaml [hunk 1]
```diff
@@ -3,6 +3,8 @@ kind: Deployment
 metadata:
   name: {{ include "datadog-operator.fullname" . }}
   namespace: {{ .Release.Namespace }}
+  annotations:
+{{ include "datadog-operator.deployment.annotations" . | indent 4 }}
```
Comment (clamoriniere) [inline charts/datadog-operator/templates/deployment.yaml:7]: for `labels` we have a helper function that compute the labels from what is provided by the user with `.Values.deployment.labels` and default labels that we always want to add.

but for annotations we don't have this kind of feature so the this line should be replace by 

```suggestion
{{ .Values.deployment.annotations" | indent 4 }}
```

**Review by clamoriniere — CHANGES_REQUESTED** [review]
the linter reviled an issue in the Deployment template.
please check my comment https://github.com/DataDog/helm-charts/pull/1449/files#r1672598611

File: charts/datadog-operator/templates/deployment.yaml (hunk 1 above)
Comment (clamoriniere) [inline charts/datadog-operator/templates/deployment.yaml:7]: for `labels` we have a helper function that compute the labels from what is provided by the user with `.Values.deployment.labels` and default labels that we always want to add.

but for annotations we don't have this kind of feature so the this line should be replace by something like:

```suggestion
{{ if .Values.deployment.annotations }}
  annotations:
{{ .Values.deployment.annotations | indent 4 }}
{{- end }}
```

Comment (clamoriniere) [reply charts/datadog-operator/templates/deployment.yaml:7]: With this change the `make update-test-baselines` doesn't panic anymore

Comment (0robustus1) [reply charts/datadog-operator/templates/deployment.yaml]: Oh, good point. Fixed.

**Comment by 0robustus1** [comment]
Comments are addressed, please take another look.

**Review by clamoriniere — APPROVED** [review]

**Pull request merged**